runtime/include/ttnn/runtime/utils.h

```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef TTNN_RUNTIME_UTILS_H
+#define TTNN_RUNTIME_UTILS_H
+
+#include "ttmlir/Target/TTNN/Target.h"
+#include "ttnn/types.hpp"
+
+namespace tt::runtime::ttnn::utils {
+
+inline ::ttnn::DataType toTTNNDataType(::tt::target::DataType dataType) {
+  switch (dataType) {
+  case ::tt::target::DataType::Float32:
+    return ::ttnn::DataType::FLOAT32;
+  // case ::tt::target::DataType::Float16:
+  //   return ::ttnn::DataType::FLOAT16;
+  case ::tt::target::DataType::BFloat16:
+    return ::ttnn::DataType::BFLOAT16;
+  case ::tt::target::DataType::UInt32:
+    return ::ttnn::DataType::UINT32;
+  case ::tt::target::DataType::UInt16:
+    return ::ttnn::DataType::UINT16;
+  // case ::tt::target::DataType::UInt8:
+  //   return ::ttnn::DataType::UINT8;
+  default:
+    throw std::runtime_error("Unsupported data type");
+  }
+}
```

Comment:
So previously we only used this function in the context of creating a host tensor, so these only cover legal host formats.  We should add all of the device formats here too if we want to use this function in device contexts too